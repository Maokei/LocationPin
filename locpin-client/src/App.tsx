import React, { useContext, useReducer } from "react";
import Button from "@material-ui/core/Button";
import { AppContext } from "./context";
import { ActionType, AppReducer } from "./reducer";
import Map from "./components/Map";

function App() {
	const initialState = useContext(AppContext);
	const [state, dispatch] = useReducer(AppReducer, initialState);
	return (
		<div className="app">
			<Button
				variant="contained"
				color="primary"
				onClick={() =>
					dispatch({ type: ActionType.USER_LOGIN, payload: "swe" })
				}>
				Hello World {state.lang}
			</Button>
			<Button
				variant="contained"
				color="primary"
				onClick={() =>
					dispatch({ type: ActionType.USER_LOGOUT, payload: "eng" })
				}>
				Reset
			</Button>
			<Map />
		</div>
	);
}

export default App;
