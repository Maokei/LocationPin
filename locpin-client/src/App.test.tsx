import React from "react";
import { render } from "@testing-library/react";

import App from "./App";

test("app component matches its snapshot", () => {
    // const { getByText } = render(<App />);
    // const linkElement = getByText(/learn react/i);
    // expect(linkElement).toBeInTheDocument();
    const app = render(<App />);
    expect(app).toMatchSnapshot();
});
