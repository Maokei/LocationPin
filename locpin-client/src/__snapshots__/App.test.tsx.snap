// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component matches its snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app"
      >
        <div
          style="outline: none;"
          tabindex="-1"
        >
          <div
            class="MuiContainer-root MuiContainer-maxWidthLg"
          >
            <div
              class="center"
              style="margin-top: 3em;"
            >
              <div
                class="MuiBox-root MuiBox-root-10 logo-border"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 1C7.59 1 4 4.59 4 9c0 5.57 6.96 13.34 7.26 13.67l.74.82.74-.82C13.04 22.34 20 14.57 20 9c0-4.41-3.59-8-8-8zm0 19.47C9.82 17.86 6 12.54 6 9c0-3.31 2.69-6 6-6s6 2.69 6 6c0 3.83-4.25 9.36-6 11.47zm-3.49-9.42v1.44h1.44l3.92-3.93-1.43-1.43zm5.88-5.34c-.15-.15-.39-.15-.54 0l-.7.7 1.44 1.44.7-.7c.15-.15.15-.39 0-.54l-.9-.9z"
                  />
                </svg>
              </div>
            </div>
            <h2
              class="center"
            >
              Log In
            </h2>
            <form
              autocomplete="off"
              class="center"
              novalidate=""
              style="padding: 0px 1em;"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-flex-end"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root input-field"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                      data-shrink="false"
                      for="input-with-icon-grid"
                      id="input-with-icon-grid-label"
                    >
                      Email Address
                    </label>
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input"
                        id="input-with-icon-grid"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <br />
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-flex-end"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12.65 10C11.83 7.67 9.61 6 7 6c-3.31 0-6 2.69-6 6s2.69 6 6 6c2.61 0 4.83-1.67 5.65-4H17v4h4v-4h2v-4H12.65zM7 14c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root input-field"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                      data-shrink="false"
                      for="input-with-icon-grid"
                      id="input-with-icon-grid-label"
                    >
                      Password
                    </label>
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input"
                        id="input-with-icon-grid"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <br />
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined btn"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Submit
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </form>
            <div
              class="MuiGrid-root link-center MuiGrid-container"
            >
              <p>
                Don't have an account?
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                  href="/signup"
                >
                  Join Now
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app"
    >
      <div
        style="outline: none;"
        tabindex="-1"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg"
        >
          <div
            class="center"
            style="margin-top: 3em;"
          >
            <div
              class="MuiBox-root MuiBox-root-10 logo-border"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 1C7.59 1 4 4.59 4 9c0 5.57 6.96 13.34 7.26 13.67l.74.82.74-.82C13.04 22.34 20 14.57 20 9c0-4.41-3.59-8-8-8zm0 19.47C9.82 17.86 6 12.54 6 9c0-3.31 2.69-6 6-6s6 2.69 6 6c0 3.83-4.25 9.36-6 11.47zm-3.49-9.42v1.44h1.44l3.92-3.93-1.43-1.43zm5.88-5.34c-.15-.15-.39-.15-.54 0l-.7.7 1.44 1.44.7-.7c.15-.15.15-.39 0-.54l-.9-.9z"
                />
              </svg>
            </div>
          </div>
          <h2
            class="center"
          >
            Log In
          </h2>
          <form
            autocomplete="off"
            class="center"
            novalidate=""
            style="padding: 0px 1em;"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-flex-end"
            >
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                  />
                </svg>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root input-field"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink="false"
                    for="input-with-icon-grid"
                    id="input-with-icon-grid-label"
                  >
                    Email Address
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      id="input-with-icon-grid"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <br />
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-flex-end"
            >
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12.65 10C11.83 7.67 9.61 6 7 6c-3.31 0-6 2.69-6 6s2.69 6 6 6c2.61 0 4.83-1.67 5.65-4H17v4h4v-4h2v-4H12.65zM7 14c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
                  />
                </svg>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root input-field"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink="false"
                    for="input-with-icon-grid"
                    id="input-with-icon-grid-label"
                  >
                    Password
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      id="input-with-icon-grid"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <br />
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined btn"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Submit
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </form>
          <div
            class="MuiGrid-root link-center MuiGrid-container"
          >
            <p>
              Don't have an account?
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="/signup"
              >
                Join Now
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
